node/rest/src/routes.rs

```diff
@@ -140,6 +140,22 @@ impl<N: Network, C: ConsensusStorage<N>, R: Routing<N>> Rest<N, C, R> {
         Ok(ErasedJson::pretty(rest.ledger.get_program(id)?))
     }
 
+    // GET /testnet3/program/{programID}/mappings
+    pub(crate) async fn get_mapping_names(
+        State(rest): State<Self>,
+        Path(id): Path<ProgramID<N>>,
+    ) -> Result<ErasedJson, RestError> {
+        Ok(ErasedJson::pretty(rest.ledger.vm().finalize_store().get_mapping_names_speculative(&id)?))
```

Comment:
Ditto. This should be `get_mapping_names_confirmed`. Working on impl now.

Reply:
I've rebased the branch rebased on the latest changes and changed the methods to the `_confirmed` variants